File: Merge_Sorted_List.py
class Node:
    def __init__(self, data):
        self.data = data
        self.link = None

def merge_sorted_lists(list1, list2):
    if list1 is None:
        return list2
    if list2 is None:
        return list1

    if list1.data <= list2.data:
        result = list1
        result.link = merge_sorted_lists(list1.link, list2)
    else:
        result = list2
        result.link = merge_sorted_lists(list1, list2.link)

    return result

File: Merge_Sorted_List.c
NODE mergeSortedLists(NODE list1, NODE list2) {
    if (list1 == NULL)
        return list2;
    if (list2 == NULL)
        return list1;

    NODE result = NULL;

    if (list1->data <= list2->data) {
        result = list1;
        result->link = mergeSortedLists(list1->link, list2);
    } else {
        result = list2;
        result->link = mergeSortedLists(list1, list2->link);
    }

    return result;
}
